tests/unit/onyxClearNativeStorageTest.js

```diff
@@ -0,0 +1,165 @@
+import AsyncStorageMock from '@react-native-async-storage/async-storage';
+import Storage from '../../lib/storage';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+
+const ONYX_KEYS = {
+    DEFAULT_KEY: 'defaultKey',
+};
+const SET_VALUE = 'set';
+const MERGED_VALUE = 'merged';
+const DEFAULT_VALUE = 'default';
+
+let storageCallResolveList = [];
+function addStorageCallResolve(name) {
+    storageCallResolveList.push(name);
+}
+
+function storageCallResolveOrder(methodName) {
+    return storageCallResolveList.indexOf(methodName) + 1;
+}
+
+let storageCallQueue = [];
+
+// Track when AsyncStorageMock.clear calls resolve.
+const asyncStorageMockClear = AsyncStorageMock.clear;
+AsyncStorageMock.clear = jest.fn(() => Promise.all(storageCallQueue)
+    .then(() => {
+        const clearPromise = asyncStorageMockClear()
+            .then(addStorageCallResolve('clear'));
+        storageCallQueue.push(clearPromise);
+        return clearPromise;
+    }));
+
+// Track when AsyncStorageMock.setItem calls resolve.
+const asyncStorageMockSetItem = AsyncStorageMock.setItem;
+AsyncStorageMock.setItem = jest.fn((key, value) => Promise.all(storageCallQueue)
+    .then(() => {
+        const setItemPromise = asyncStorageMockSetItem(key, value)
+            .then(addStorageCallResolve('setItem'));
+        storageCallQueue.push(setItemPromise);
+        return setItemPromise;
+    }));
+
+describe('Set data while storage is clearing', () => {
+    let connectionID;
+    let Onyx;
+    let onyxValue;
+
+    /** @type OnyxCache */
+    let cache;
+
+    beforeAll(() => {
+        Onyx = require('../../index').default;
```

Comment:
Why not just do this at the top of the file?

Reply:
I put that there because I am mocking `AsyncStorage.clear` and `AsyncStorage.setItem`, and I thought that I had to do that before requiring Onyx. I put it at the top of the file and tested it, and I was wrong.

Reply:
I initially used this type of expressions for one of my tests

Re-requiring Onyx gives us a fresh instance (empty cache, no connections etc...) 
It makes sense to `require` like that in a `beforeEach` blocks, but not in `beforeAll`